tests/framework/randomness/test_crn.py

```diff
@@ -0,0 +1,301 @@
+            additional_key="attr1",
+        )
+        self.population_view.update(population)
+
+
+class SequentialPopulation(BasePopulation):
+    """Population that bases identity on the order simulants enter the simulation.
+
+    NOTE: This population is not fully supported by the CRN system and is here to explicitly
+    test and assert the expected failure cases.
+    """
+
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        self.count = 0
+
+    def on_initialize_simulants(self, pop_data: SimulantData):
+        new_people = len(pop_data.index)
+
+        population = pd.DataFrame(
+            {
+                "crn_attr1": pd.Timestamp("2020-01-01"),
+                "crn_attr2": range(self.count, self.count + new_people),
+            },
+            index=pop_data.index,
+        )
+
+        if self.with_crn:
+            self.register(population)
+
+        population["other_attr1"] = self.randomness_other.get_draw(
+            pop_data.index,
+            additional_key="attr1",
+        )
+        self.population_view.update(population)
+        self.count += new_people
+
+
+@pytest.mark.parametrize(
+    "pop_class, with_crn, sims_to_add",
+    [
+        pytest.param(EntranceTimePopulation, True, cycle([0])),
+        pytest.param(EntranceTimePopulation, True, cycle([2])),
+        pytest.param(EntranceTimePopulation, False, cycle([0])),
+        pytest.param(EntranceTimePopulation, False, cycle([2])),
+        pytest.param(SequentialPopulation, True, cycle([0])),
+        pytest.param(SequentialPopulation, True, cycle([2])),
+        pytest.param(SequentialPopulation, False, cycle([0])),
+        pytest.param(SequentialPopulation, False, cycle([2])),
+    ],
+)
+def test_multi_sim_basic_reproducibility_with_same_pop_growth(
+    pop_class,
+    with_crn,
+    sims_to_add,
+):
+    if with_crn:
+        configuration = {"randomness": {"key_columns": ["crn_attr1", "crn_attr2"]}}
+    else:
+        configuration = {}
+
+    sim1 = InteractiveContext(
+        components=[pop_class(with_crn=with_crn, sims_to_add=sims_to_add)],
+        configuration=configuration,
+    )
+    sim2 = InteractiveContext(
+        components=[pop_class(with_crn=with_crn, sims_to_add=sims_to_add)],
+        configuration=configuration,
+    )
+
+    pop1 = sim1.get_population().set_index(["crn_attr1", "crn_attr2"])
+    pop2 = sim2.get_population().set_index(["crn_attr1", "crn_attr2"])
+    assert_frame_equal(pop1, pop2)
+
+    for i in range(2):
+        sim1.step()
+        sim2.step()
+
+    pop1 = sim1.get_population().set_index(["crn_attr1", "crn_attr2"])
+    pop2 = sim2.get_population().set_index(["crn_attr1", "crn_attr2"])
+    assert_frame_equal(pop1, pop2)
+
+
+@pytest.mark.parametrize(
+    "pop_class, with_crn",
+    [
+        pytest.param(EntranceTimePopulation, True),
+        pytest.param(EntranceTimePopulation, False),
+        pytest.param(SequentialPopulation, True, marks=pytest.mark.xfail),
```

Comment:
This is just marking that the test is expected to fail. Is there a way to mark that we expect it to fail on a specific assert statement? By this is mean, we expect the first `assert_frame_equal` call to pass and the later call to fail.

Reply:
Yeah, I can write a more granular test.  I'll think about it.